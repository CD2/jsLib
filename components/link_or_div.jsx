import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

export class LinkOrDiv extends React.Component {

  static propTypes = {
    noLink: PropTypes.bool
  }

  render() {
    if (this.props.noLink) {
      const props = { ...this.props }
      delete props.href
      delete props.to
      delete props.noLink
      return (<div {...props} />)
    }
    const props = { ...this.props }
    delete props.noLink
    return (<Link {...props} />)
  }

}
export default LinkOrDiv
