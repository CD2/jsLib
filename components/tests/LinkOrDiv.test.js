/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { LinkOrDiv } from '../link_or_div'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe(`<LinkOrDiv />`, () => {
  const view = shallow((
    <LinkOrDiv />
  ))

  it(`snapshot`, ()=>expect(toJson(view)).toMatchSnapshot())
})