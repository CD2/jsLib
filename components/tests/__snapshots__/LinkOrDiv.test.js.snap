// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkOrDiv /> snapshot 1`] = `
<Link
  replace={false}
/>
`;

exports[`snapshot 1`] = `
<Link
  replace={false}
/>
`;
